libsolidity/formal/SolverInterface.h

```diff
@@ -65,6 +65,28 @@ class Expression
 	Expression& operator=(Expression const&) = default;
 	Expression& operator=(Expression&&) = default;
 
+	/// Checks whether this operator has a valid arity based on its name.
+	/// _arity is the arity of the operator in the solver.
+	bool isValid(unsigned _arity) const
```

Comment:
Shouldn't this be only dependant on the internal state (`m_arguments.size()`)?

Reply:
It is indirectly. Z3 and CVC4 interfaces use the `_expr.arguments` to build the correspondent vector of arguments. It _should_ be the same in the end, it just checks when more thing this way

Reply:
I'm not sure if it is better this way or to have two different assertions. I think this `isValid` would need to be local to the `Expression` class.

Reply:
Makes sense. I'll make it two assertions then, to keep it safe

Reply:
Actually I don't think we need the second assertion.